src/hotspot/share/opto/graphKit.cpp

```diff
@@ -1067,7 +1067,9 @@ bool GraphKit::compute_stack_effects(int& inputs, int& depth) {
     {
       bool ignored_will_link;
       ciField* field = method()->get_field_at_bci(bci(), ignored_will_link);
-      int      size  = field->type()->size();
+      int size = field->is_multifield_base() ?
```

Comment:
Encapsulate size computation for mulitfields in `ciField`?

Reply:
Decision to scalarize a multi-field is delegated to compiler, ci* model is sacrosanct.  So a size query over ciMultiField (base) will always return the entire bundle size, we have to to a special handling on compiler side. I have factored this into a static routine.